### ISSUE TITLE
support blocking packages

### ISSUE THREAD

**Issue by ktdreyer** [ktdreyer/koji-ansible#2]
Here is how to define a package set for a tag today:

```yaml
  - name: Create a main product koji tag
    koji_tag:
      name: ceph-3.2-foo
      state: present
      packages:
        kdreyer:
          - ansible
          - ceph
          - ceph-ansible
```

This means I can only "whitelist" (`koji add-pkg`) builds. I need to be able to "blacklist" them (`koji block-pkg`).

Need to play around with whether it makes sense to build this into the `packages` parameter. Something like:
```yaml
      packages:
        kdreyer:
          - ansible
          - ceph:
              blocked: true
          - ceph-ansible
```

(This would also make room for the `extra_arches` feature of `packageListAdd`.)

**Comment by mizdebsk** [comment]
I'm hitting this issue too, so I'll probably implement and submit a fix soon.

**Comment by ktdreyer** [comment]
Now that we're using `koji_tag` in some large environments and admins have manually blocked certain packages, we need some integration tests here to ensure backwards compatibility.

### Current behavior that we must preserve:

**State:** Package is missing from a tag
**Action:** A package name string
**Expected result:** Ansible adds package and does not block package.

**State:** Package is present in a tag and not blocked
**Ansible:** package name is absent
**Expected result:** Ansible removes package from tag

**State:** Package is present in a tag and blocked
**Ansible:** package name is absent
**Expected result:** Ansible removes package from tag (_I've not tested this!_)

And most importantly, idempotency for bare strings:

**State:** Package is present in a tag and not blocked
**Ansible:** a package name string
**Expected result:** Ansible does nothing (package is still unblocked)

**State:** Package is present in a tag and blocked
**Ansible:** a package name string
**Expected result:** Ansible does nothing (package is still blocked)


### New behavior to add:

**State:** Package is missing from a tag
**Action:** A package name dict with `blocked: true`
**Expected result:** Ansible adds package and blocks package.

**State:** Package is missing from a tag
**Action:** A package name dict with `blocked: false`
**Expected result:** Ansible adds package and does not block package.

**State:** Package is present in a tag and not blocked
**Action:** A package name dict with `blocked: true`
**Expected result:** Ansible blocks package in tag

**State:** Package is present in a tag and blocked
**Action:** A package name dict with `blocked: false`
**Expected result:** Ansible unblocks package in tag

For completeness, we should test these additional idempotency scenarios

**State:** Package is present in a tag and not blocked
**Action:** A package name dict with `blocked: false`
**Expected result:** Ansible does nothing (package is still not blocked)

**State:** Package is present in a tag and blocked
**Action:** A package name dict with `blocked: true`
**Expected result:** Ansible does nothing (package is still blocked)

**Comment by ktdreyer** [comment]
This is resolved #228 for `koji_tag_packages` and in #242 for `koji_tag`.
